airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java

```diff
@@ -729,6 +737,37 @@ public void setSortAlbumsByYear(boolean b) {
         setBoolean(KEY_SORT_ALBUMS_BY_YEAR, b);
     }
 
+    public boolean getIgnoreSymLinks() {
+        return getBoolean(KEY_IGNORE_SYMLINKS, DEFAULT_IGNORE_SYMLINKS);
+    }
+
+    public void setIgnoreSymLinks(boolean b) {
+        setBoolean(KEY_IGNORE_SYMLINKS, b);
+    }
+
+    public String getExcludePatternString() {
+        return getString(KEY_EXCLUDE_PATTERN_STRING, DEFAULT_EXCLUDE_PATTERN_STRING);
+    }
+
+    public void setExcludePatternString(String s) {
+        setString(KEY_EXCLUDE_PATTERN_STRING, s);
+        compileExcludePattern();
+    }
+
+    private void compileExcludePattern() {
+        if (getExcludePatternString() != null && getExcludePatternString().trim().length() > 0) {
+            excludePattern = Pattern.compile(getExcludePatternString());
+            excludePatternCompiled = true;
+        }
+    }
+
+    public Pattern getExcludePattern() {
+        if (! excludePatternCompiled) {
+            compileExcludePattern();
```

Comment:
Can't this be simplified to just a null check on excludePattern? The excludePatternCompiled boolean is redundant as currently written. One valid reason to keep the flag is if you set it in the case the pattern string is null or there is an error compiling it so as to avoid unnecessarily recompiling, but that would require changes to compileExcludePattern.